Parsers/SysmonParser.txt

```diff
@@ -0,0 +1,238 @@
+// KQL Sysmon Event Parser
+// Date: May 8, 2019
+// Notes:
+// 1. This parser works again the current version of events from Sysmon, it may need updates if Sysmon is updated.
+// 2. technique_id and technique_name will only be parsed/available if deployed via a custom XML
+//
+let timeframe = ago(1d);
+let EventData = Event
+| where TimeGenerated >= timeframe
+| where Source == "Microsoft-Windows-Sysmon"
+| extend RenderedDescription = tostring(split(RenderedDescription, ":")[0]) //All
+| project TimeGenerated, Source, EventID, Computer, UserName, EventData, RenderedDescription
+| extend EvData = parse_xml(EventData)
+| extend p = EvData.DataItem.EventData.Data
+| project-away EventData, EvData
+;
+let SysmonEvent1=() {
+let processEvents = EventData
+| where EventID == 1
+| extend RuleName = p.[0].["#text"], UtcTime = p.[1].["#text"], ProcessGuid = p.[2].["#text"], ProcessId = p.[3].["#text"], Image = p.[4].["#text"],
+FileVersion = p.[5].["#text"], Description = p.[6].["#text"], Product = p.[7].["#text"], Company = p.[8].["#text"],
+CommandLine = p.[9].["#text"], CurrentDirectory = p.[10].["#text"], User = p.[11].["#text"], LogonGuid = p.[12].["#text"],
+LogonId = p.[13].["#text"], TerminalSessionId = p.[14].["#text"], IntegrityLevel = p.[15].["#text"], Hashes = p.[16].["#text"], 
+ParentProcessGuid = p.[17].["#text"], ParentProcessId = p.[18].["#text"], ParentImage = p.[19].["#text"], ParentCommandLine = p.[20].["#text"]
+| parse RuleName with * 'technique_id=' TechniqueId ',' * 'technique_name=' TechniqueName 
+| parse Hashes with * 'SHA1=' SHA1 ',' * 'MD5=' MD5 ',' * 'SHA256=' SHA256 ',' * 'IMPHASH=' IMPHASH 
+| project-away p, RuleName, Hashes
+;
+processEvents;
+};
+let SysmonEvent2=() {
```

Comment:
suggest we name the functions more informatively - eg SysmonEvent1_ProcessCreate, SysmonEvent2_ReadProcessMemory, etc?